### PR TITLE
Fixes #798 : Accepts Manual Data in API

### LINKED ISSUE #798: API for accepting manual data
**I'm submitting a ...**
- [ ] bug report
- [X] feature request

Create an API endpoint to accept, sanitize and store the manual data as a CSV so that it can be used in future for generating badges and other purpose.

**Yes, I am working on it and will send a PR for the same**

### PR DESCRIPTION
**Pull request by gabru-md**
Fixes #798 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [ ] I have added necessary documentation (if appropriate)

### Preview Link 

- **Replace XXX with your PR no**
- Link to live demo: http://pr-XXX-fossasia-badgeyay.surge.sh  

#### Changes proposed in this pull request:

- Adds API endpoints for accepting Manual Data

CC @djmgit

### CODE REVIEW

**Comment by gabru-md** [comment]
@ParthS007 @yashLadha please review

**Comment by gabru-md** [comment]
@ParthS007 @yashLadha please review

File: backend/blueprint/api/controllers/fileUploader.py [hunk 1]
```diff
@@ -59,3 +59,36 @@ def fileUpload():
...
+    manual_data = data.get('manual_data')
+    try:
+        csvName = saveAsCSV(csvData=manual_data)
+    except Exception as e:
+        return(jsonify(
+            Response(401).exceptWithMessage(
+                str(e),
+                'Manual Data could not be uploaded')))
+
+    file_upload = File(csvName, 'csv')
```
Comment (djmgit) [inline backend/blueprint/api/controllers/fileUploader.py]: File model has User.id as foreign key. Presently no user id is being specified (ideally it should be the logged in user). So what default value are we using?

**Comment by gabru-md** [comment]
@djmgit I am sorry for replying up late.
I figured out that what you pointed out was correct.
Therefore I have pushed another commit which compensates all the previous errors that have been happening but have been ignored or were unknown.

My changes are tested and they work great.
Let me list down the changes I've done:
- [ ] created a route for manual_data upload
- [ ] corrected the `File` database model
- [ ] added proper code for relating to the correct uploader
- [ ] fixing bugs in `registerUser.py`
- [ ] creating and fixing the `saveAsCSV` function
- [ ] corrects the `User` database model
- [ ] modify `getUser` function in `User` model
- [ ] commented the `sendMail` part since we are using `firebase` for the same

I know that travis might be failing along with codacy, but this PR works perfectly fine. from `registering user` to `login-in` and then to uploading `manual_data` to the correct api.

I will resolve the Codacy and Travis errors, in the meanwhile can you @ParthS007 and @yashLadha please review this PR.

**Comment by gabru-md** [comment]
Here is a video to help you review that my work is correct and works without flaw.
[Video Preview](https://www.youtube.com/watch?v=a4RrFzfA-cc&feature=youtu.be)

Thank You

**Comment by gabru-md** [comment]
I am also posting the database screenshots to show that the database entries also work perfectly fine.

s1
The above image is the two tables in the database.


s2
The above image is of the name of file uploaded with the corresponding `user_id`


s3
The above image is of the user stored in the postgresql server in the backend.

The video shows the complete process and the csv uploaded.
Thank You

CC : @djmgit @harshit98 @ParthS007 @yashLadha

**Comment by djmgit** [comment]
@gabru-md I know github provides an option to squash merge, but it is an humble request to please send one commit per PR as it helps in review.
This is my request to everyone @yashLadha @ParthS007 @vaibhavsingh97 
@gabru-md please squash the commits into one.

**Comment by gabru-md** [comment]
@djmgit the work continued late in night, so once it was all done I turned down my laptop and went to sleep.
I was going to squash them. There are multiple commits due to the fact that there were many times that travis and codacy etc were failing so I just send multiple commits to solve them all.
I'll squash them in a moment. Please review it.

**Pull request merged**